packages/protocol/contracts/stability/StableToken.sol

```diff
@@ -132,21 +132,32 @@ contract StableToken is
    * @param rate new rate.
    * @param updatePeriod how often inflationFactor is updated.
    */
-  function setInflationParameters(uint256 rate, uint256 updatePeriod)
-    external
-    onlyOwner
-    updateInflationFactor
-  {
+  function setInflationParameters(uint256 rate, uint256 updatePeriod) external onlyOwner {
+    /* solhint-disable not-rely-on-time */
     require(rate != 0, "Must provide a non-zero inflation rate.");
-    inflationState.rate = FixidityLib.wrap(rate);
-    inflationState.updatePeriod = updatePeriod;
 
-    emit InflationParametersUpdated(
-      rate,
-      updatePeriod,
-      // solhint-disable-next-line not-rely-on-time
-      now
+    uint256 completePeriodsLastUpdated;
+    (inflationState.factor, completePeriodsLastUpdated) = getUpdatedInflationFactor();
```

Comment:
you can keep the `updateInflationFactor` modifier and skip this step

Reply:
or if you want to save gas, write intermediate factor to a `memory` variable, instead of `storage`

Reply:
wanted to avoid multiple `InflationFactorUpdated` events but maybe it's useful to see the intermediate change

Reply:
oh. I've missed that. Yeah, in that case better to avoid it then.
Still you can save was by using memory intermediate variable